timer/timer.go

```diff
@@ -0,0 +1,144 @@
+package timer
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// Timer definition
+type (
+	Timer struct {
+		ID             uuid.UUID
+		Domain         string
+		ExecutionCount uint32
+		Meta           Meta
+		Task           Task
+		Schedule       Schedule
+	}
+
+	// Task represents a timer task.
+	Task interface {
+		isTask()
```

Comment:
This is weird - the interface is named task and it has one method isTask? Not clear.